contributing/development/cpp_usage_guidelines.rst

```diff
@@ -53,18 +53,69 @@ as Godot provides its own data types (among other things).
+|                        |                          | preferred.                                                                          |
++------------------------+--------------------------+-------------------------------------------------------------------------------------+
+| ``Array`` 📜           | ``std::vector``          | Values can be of any Variant type. No static typing is imposed.                     |
++------------------------+--------------------------+-------------------------------------------------------------------------------------+
+| ``TypedArray`` 📜      | ``std::vector``          | Equivalent to ``Array`` but with static typing for its elements.                    |
+|                        |                          | Not to be confused with ``Packed*Array``, which is internally a ``Vector``.         |
++------------------------+--------------------------+-------------------------------------------------------------------------------------+
+| ``Packed*Array`` 📜    | ``std::vector``          | Equivalent to ``Vector``, e.g. ``PackedColorArray = Vector<Color>``.                |
+|                        |                          | Only a limited list of packed array types are available                             |
+|                        |                          | (use ``TypedArray`` otherwise).                                                     |
++------------------------+--------------------------+-------------------------------------------------------------------------------------+
+| ``Span``               | ``std::span``            | Represents read-only access to a contiguous array without needing to copy any data. |
+|                        |                          | See `pull request description <https://github.com/godotengine/godot/pull/100293>`__ |
+|                        |                          | for details.                                                                        |
++------------------------+--------------------------+-------------------------------------------------------------------------------------+
+| ``HashSet``            | ``std::set``             | **Use this as the "default" set type.**                                             |
++------------------------+--------------------------+-------------------------------------------------------------------------------------+
+| ``RBSet``              | ``std::set``             | Uses a `red-black tree <https://en.wikipedia.org/wiki/Red-black_tree>`__            |
+|                        |                          | for faster access.                                                                  |
++------------------------+--------------------------+-------------------------------------------------------------------------------------+
+| ``HashMap``            | ``std::map``             | **Use this as the "default" map type.** Iteration is different from ``std::map``    |
```

Comment:
I'd argue that the relevant difference isn't the ordering, but the keying, `std::map` is a key based system, which is mostly relevant for things like time complexity, so I'd say this should be `std::unordered_map`